gobblin-runtime/src/main/java/gobblin/runtime/Limiter.java

```diff
@@ -0,0 +1,119 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime;
+
+
+/**
+ * An interface for classes that implement some logic limiting on the occurrences of some events,
+ * e.g., data record extraction using an {@link gobblin.source.extractor.Extractor}.
+ *
+ * @author ynli
+ */
+public interface Limiter {
+
+  /**
+   * Supported types of {@link Limiter}s.
+   */
+  public enum Type {
```

Comment:
I don't think the Limiter interface should be limiting the possible implementations :)